File: typescript/src/problems1_100/merge-sorted-array.ts
class Problem {
  merge(nums1: number[], m: number, nums2: number[], n: number) {
    throw new Error(`Not implemented (${nums1}, ${m}, ${nums2}, ${n}).`);
  }
}

class Default extends Problem {
  merge(nums1: number[], m: number, nums2: number[], n: number) {
    if (n >= 0) {
      for (let i = 0; i < n; i++) {
        nums1[m + i] = nums2[i];
      }
    }
    nums1.sort((a, b) => a - b);
  }
}

const SOLUTIONS = [new Default()];
export default SOLUTIONS;

File: python/src/problems1_100/merge_sorted_array.py
from abc import ABC, abstractmethod
from typing import override


class Problem(ABC):
    @abstractmethod
    def merge(self, nums1: list[int], m: int, nums2: list[int], n: int):
        pass


class Default(Problem):
    @override
    def merge(self, nums1: list[int], m: int, nums2: list[int], n: int):
        if n >= 0:
            nums1[m:m + n] = nums2[0:n]
        nums1.sort()


SOLUTIONS: list[Problem] = [Default()]
